RViz may crash when TF's childFrameId is " "

When the childframeid of TF is ' ' and subscribed, RViz will crash.
![bug](https://user-images.githubusercontent.com/24313254/151276459-ea17a0ef-0c29-48a0-9bf7-a0f2c4e72a19.png)


When the childframeid of TF is '\a'  or '\e' or other strings starting with' \ ',Abnormal display of RViz.
![Abnormal display](https://user-images.githubusercontent.com/24313254/151276544-5c618bad-8a26-491f-b574-9fa60ec70515.png)


### Your environment
* OS Version: e.g. Ubuntu 18.04
* ROS Distro: [Melodic]
* RViz, Qt, OGRE, OpenGl version as printed by rviz:

![bug](https://user-images.githubusercontent.com/24313254/151276494-9cb04804-cc3c-4cad-b646-0295b0a8f781.png)

I cannot confirm these issues:
TF already rejects/ignores an empty child transform with the following error message on console:
`[ERROR] ros.rosconsole_bridge.TF_NO_CHILD_FRAME_ID: Ignoring transform from authority "unknown_publisher" because child_frame_id not set`

When sending a child frame with a non-printable control character in its name (e.g. `\a`), what exactly do you expect to happen?

> I cannot confirm these issues: TF already rejects/ignores an empty child transform with the following error message on console: `[ERROR] ros.rosconsole_bridge.TF_NO_CHILD_FRAME_ID: Ignoring transform from authority "unknown_publisher" because child_frame_id not set`
> 
> When sending a child frame with a non-printable control character in its name (e.g. `\a`), what exactly do you expect to happen?

It's not empty, it's a space.
When sending a child frame with a non-printable control character in its name (e.g. \a), I expect it to remain '\ a' without escape.

--- a/src/rviz/ogre_helpers/movable_text.cpp
+++ b/src/rviz/ogre_helpers/movable_text.cpp
@@ -244,7 +244,7 @@ void MovableText::_setupGeometry()
     mUpdateColors = true;
   }
 
-  if (utfCaption.empty())
+  if (vertexCount == 0)
   {
     return;
   }

